pkg/registry/apis/provisioning/file.go

```diff
@@ -0,0 +1,211 @@
+package provisioning
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"path/filepath"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	"k8s.io/apiserver/pkg/registry/rest"
+
+	"github.com/grafana/grafana/pkg/apimachinery/apis/common/v0alpha1"
+	provisioning "github.com/grafana/grafana/pkg/apis/provisioning/v0alpha1"
+)
+
+type readConnector struct {
+	getter RepoGetter
+}
+
+func (*readConnector) New() runtime.Object {
+	// This is added as the "ResponseType" regardless what ProducesObject() returns
+	return &provisioning.ResourceWrapper{}
+}
+
+func (*readConnector) Destroy() {}
+
+func (*readConnector) NamespaceScoped() bool {
+	return true
+}
+
+func (*readConnector) GetSingularName() string {
+	return "Resource"
+}
+
+func (*readConnector) ProducesMIMETypes(verb string) []string {
+	return []string{"application/json"}
+}
+
+func (*readConnector) ProducesObject(verb string) any {
+	return &provisioning.ResourceWrapper{}
+}
+
+func (*readConnector) ConnectMethods() []string {
+	return []string{http.MethodGet, http.MethodPost, http.MethodDelete}
```

Comment:
🤔  maybe we should do PUT also, so there is a clear CREATE vs UPDATE 🤔 

And the settings could be:
```
type EditingOptions struct {
	Create bool `json:"create"`
	Update bool `json:"update"`
	Delete bool `json:"delete"`
}
```